webui/src/components/editrecord.jsx

```diff
@@ -359,7 +359,7 @@ const EditRecord = React.createClass({
                     {!title ? false :
                         <label htmlFor={id} className="col-sm-3 control-label" style={{fontWeight:'bold'}}>
                             <span style={{float:'right', color:isError?'red':'black'}}>
-                                {title} {schema.get('isRequired') ? "*":""}
+                                {title} {schema.get('isRequired') && (!this.props.isDraft || (this.props.isDraft && this.isForPublication())) ? "*":""}
```

Comment:
We could always show the stars to indicate the required fields, I'm not sure what's the best solution here, do as you wish.

The boolean test can be simplified: `schema.get('isRequired') && (!this.props.isDraft || this.isForPublication())`